programs/svm-spoke/src/instructions/slow_fill.rs

```diff
@@ -131,12 +134,15 @@ pub struct ExecuteV3SlowRelayLeaf<'info> {
     #[account(seeds = [b"root_bundle", state.key().as_ref(), root_bundle_id.to_le_bytes().as_ref()], bump)]
     pub root_bundle: Account<'info, RootBundle>,
 
+    /// CHECK: This is not dangerous as we don't read or write from this account. Passed in as a prop for the constraint.
+    pub relay_hash: UncheckedAccount<'info>,
+
     #[account(
         mut,
-        seeds = [b"fills", relay_hash.as_ref()],
+        seeds = [b"fills", relay_hash.key().as_ref()],
         bump,
         // Make sure caller provided relay_hash used in PDA seeds is valid.
-        constraint = is_relay_hash_valid(&relay_hash, &slow_fill_leaf.relay_data, &state) @ SvmError::InvalidRelayHash
+        constraint = is_relay_hash_valid(&relay_hash.key().as_ref().try_into().unwrap(), &slow_fill_leaf.relay_data, &state) @ SvmError::InvalidRelayHash
```

Comment:
What happens if we pass an account that fails the `try_into()` conversion, i.e. cannot be unwrapped? Does it throw `SvmError::InvalidRelayHash`, or should we handle it explicitly with something like `.as_ref().try_into().map_err(|_| ...`

Reply:
good question. will test it and report back.

Reply:
```
try {
      const invalidRelayHash = new PublicKey("InvalidPublicKeyString");
      await program.methods
        .fillV3Relay(relayData, new BN(1), relayer.publicKey)
        .accounts({
          ...accounts,
          relayHash: invalidRelayHash,
        })
        .signers([relayer])
        .rpc();
      assert.fail("Should not be able to fill relay with an invalid relayHash");
    } catch (err: any) {
      console.log("Error with non-correct public key relayHash:", err.toString());
    }

    // Test with a relayHash that is an invalid public key
    try {
      const invalidRelayHash = new PublicKey(new Uint8Array(32).fill(255)); // Invalid public key
      await program.methods
        .fillV3Relay(relayData, new BN(1), relayer.publicKey)
        .accounts({
          ...accounts,
          relayHash: invalidRelayHash,
        })
        .signers([relayer])
        .rpc();
      assert.fail("Should not be able to fill relay with an invalid relayHash");
    } catch (err: any) {
      console.log("Error with invalid public key relayHash:", err.toString());
    }
```
returns:
```
Error with non-correct public key relayHash: Error: Non-base58 character
Error with invalid public key relayHash: AnchorError caused by account: fill_status. Error Code: ConstraintSeeds. Error Number: 2006. Error Message: A seeds constraint was violated.
```
in either case we don't actually get the expected error, rather the more cryptic issue on seed constrain violation, which is a bit painful. this makes this a bit worse than the previous implementation as it kinda hides the errors, which is less good.

WDYT?

Reply:
a bit more info on this...I played with it some more tonight and _we generate the same errors with the previous implementation_. i.e we did not loose error quality.

Reply:
Ok, I think this solution is good IMO, I would just add a @dev comment explaining why we are doing this.